### ISSUE TITLE
SC 2.4.13  Page break navigation for Web content: overly prescriptive?

### ISSUE THREAD

**Issue by spanchang** [w3c/wcag#2026]
I believe this requirement is being overly prescriptive.
How is a "page break locator"  different from other structural identifier like a section, landmark, headinmg, table, etc.?
For example , suppose the instructor  in the class says, "look at table 2". Should there be an SC that explicitly says,   
"For web content with data tables, a mechanism is available to navigate to each data table".
Should there be an SC that  says  something similar for section identifiers and headings so a user can follow the instructor's guidance, "go to Chapter 17"? And sometimes in different editions,  chapter numbers also get reorganized. 

No question digital publications and learning content  is ubiquitous. 
Consider someone guiding a  group of individuals  on filling out a tax return with one accessing a fillable PDF form, one accessing a paper tax return and a third using a Web form.
For a printed form there is a text label next to the text box so SC 1.3.1 needs programmatic association in digital content.

As the Understanding doc notes, Statically paginated formats (e.g., PDF)  and EPUB  provide a page navigation feature. These formats are designed  to duplicate the  printed formats in a  digital fashion unlike HTML content.
So first, HTML (Web) content  needs to have a mechanism of page numbers that align with the specified edition (like EPUB) and meet this digital publication requirement (mentioned in the Understanding doc). 
Thereafter, for "content with page break locators", true the  page break locator needs to be programmatically determinable, identifiable and  navigable just like other structural indicators.
As all of us know, when authored functionality requires it, Web pages do have the ability to permit pagination controls  to be presented  so users can navigate to a desired page and in fact often permit users to set how many rows should be displayed per page.
So when content is authored  with page numbering , pagination controls are made navigable by authors. If they are not they fail functionality mandates for the content.  Existing WCAG SCs are   sufficient to flag any accessibility issues that may be present with these elements.   
Another point: perhaps instructors should provide alternate cues for  finding content. And when one is trying to find content based on a reference  in a footnote or bibliography, the lack of alignment across formats and editions is problematic for all users.   
The requirement makes sense for EPUB  as in
http://idpf.org/epub/a11y/#sec-page-nav
but not for a Web page.
An HTML Web page does not have headers / footers that may contain the page# like a PDF or EBUB or Word doc.
Even https://www.w3.org/WAI/WCAG21/Techniques/pdf/PDF17 references SC 1.3.1.
The need for this SC needs a re-look. 
Thanks,
Sailesh

**Comment by alastc** [comment]
Proposed response:

> How is a "page break locator" different from other structural identifier like a section, landmark, headinmg, table, etc.?

From the linked definition: "programmatically determinable destination markers that are arranged in a meaningful sequence to determine the location of a page in relation to others in the set."

Sections, landmarks, headings etc do not determine the location of pages in a set.

> Should there be an SC that says something similar for section identifiers and headings so a user can follow the instructor's guidance

That is not the use-case this criterion is trying to address. If you'd like to suggest that requirement, we could discuss that for a future version.

> Existing WCAG SCs are sufficient to flag any accessibility issues that may be present with these elements.

There is no current requirement to provide navigation to page break locators. 

From your comments, it seems you have not come across the core use-case which this SC addresses. There are HTML (in browser) renderings of e-books which do not have a navigation mechanism to get to the page references.

This scenario was brought to the AG group by the same people who wrote the epub spec that you linked to, and they made a good case that this is a scenario that needs addressing. (There are other scenarios as well, but this was the one that could be addressed most quickly in WCAG.) Given their expertise in the domain, the group are satisfied that there is a need for this requirement.

**Comment by spanchang** [comment]
The proposed SC addresses "For web content with page break locators" ... which to me sounds like Web content that already have  these mechanisms in place available generally but not in an accessible manner.  
The use-case you highlight on the other hand, "There are HTML (in browser) renderings of e-books which do not have a navigation mechanism to get to the page references" suggests that the mechanisms are completely absent. In other words the ability to navigate to different pages is not available for anyone disabled or not. Am I reading this correctly?
So is the SC suggesting that "page locators" be added, i.e. new functionality be introduced into the content?
Surely this cannot be  reasonable.
Yes, if navigation mechanisms are present but are not accessible, then existing SCs can be applied to make these mechanisms accessible.
The use-case rightly  refers to these as "navigational mechanisms" which are no different from  such mechanisms one sees  across all Web content: main menus / secondary or left nav menus or pagination  links present for displaying transaction rows etc.  
So why is a new term needed , page break locators for  digital books?
In my understanding an SC that requires functionality to be added by inserting new content like SC 2.4.10, "Section headings are used to organize the content" is Level AAA. 
Indeed I am no expert in digital publication as you rightly observe, but a PDF or EPUB is   a very specific type of digital content designed for e-books and such. HTML is a general purpose  Web technology that can be used imaginatively for  a variety of applications. 
So if this new SC is needed why is it not Level AAA?
Thanks for your attention and time.
Sailesh

**Comment by alastc** [comment]
> The use-case you highlight on the other hand, "There are HTML (in browser) renderings of e-books which do not have a navigation mechanism to get to the page references" suggests that the mechanisms are completely absent. In other words the ability to navigate to different pages is not available for anyone disabled or not. Am I reading this correctly?

I think in most cases there are visual indicators as well as programmatic ones, for example, there is a "Page number" displayed at the bottom of the Kindle online reader. Depending on your text-size level you might have multiple screens per 'page number'. It is part of the footer, and clicking on it switches between page and location.

As someone who can see it easily, I can scrub through the screens quickly to find the right page. However, if you are using assistive technology, that is not easy.

However, in the top bar there is a feature to enter a page number and go straight there.


> So is the SC suggesting that "page locators" be added, i.e. new functionality be introduced into the content?

No, that was going to be a separate SC, but it was difficult to pin down how you would assess whether content should be required to have it. (It would involve comparing to offline versions!)

This SC is saying that: If you include mark-up to programmatically mark where pages start, there should be a mechanism to navigate to them.


> So why is a new term needed , page break locators for digital books?

Regular web pages do not have such markup. Therefore it is only really applicable to people who are putting ePub style books online. 


> In my understanding an SC that requires functionality to be added by inserting new content like SC 2.4.10, "Section headings are used to organize the content" is Level AAA.

This isn't asking for changes to the primary content of the page, it is asking for a navigation mechanism (more like multiple ways). Any book type of content will have some form of navigation, so this is ensuring that the page-navigation is included.

**Comment by spanchang** [comment]
When one uses tools to convert  digital book formats from PDF or such into  HTML, it would  be extra work to provide ability in HTML content to navigate to pages.
Screen reader users can search for  "63" if they are looking for page #63 in HTML. It is possible "63" could also be in the content and it would be difficult in those cases.
If  I see there is a logo image in the footer or  some similar artefact / text content in the footer or a separator, I can search for that text or navigate by image / jump from one  separator to the next.  Something like what a sighted user will do to go from one page to the next in HTML.
The toolbar  feature to jump to a page is the user agent feature of a digital book reader. 
EPUB and PDFs are apps  meant for  reading books and they provide  page navigation options. So there is an onus on users too to use the proper app based on content.
Consider a spreadsheet content with formulae that recalculate data based on a change to some other cell.         If a user were to copy that table into Microsoft Word he/she cannot expect that same functionality, right? One cannot require Microsoft to implement formulas and recalculation functionality in Word.
SC 2.4.5 Multiple Ways is for a website (set of pages) and that too is at Level AA. As findability and navigation are important for a website, SC 2.4.5 does require a search method or table of Contents or Sitemap to be added. Yes that is functionality / content being mandated by an SC. 
Adding SCs like 2.4.13  and defining new terms  simply makes  WCAG more complex, overly prescriptive and daunting.
If one were to consider this SC and some others in WCAG 2.1 or proposed 2.2 as part of the original set in WCAG 2.0, it is likely some would not have made it to the final list. Or, they would be Level AAA at best.
With that I rest my case.
Thanks,
Sailesh

**Comment by mraccess77** [comment]
This SC only applies when the page numbers are communicated as programmatic markers - so if they were just in text then I'm not sure how it would apply.  I would like to know what is meant by programmatic markers in this case as that makes a big difference to whether this SC is in scope.

As an aside, Just searching for a page number isn't always helpful especially when documents have numbers in them.   When any page number is display in my opinion it should be then required to be programmatically available.

I often read content that was from print in HTML and page numbers are generally there but just random text in between paragraphs with no context or no communication that they are actually page numbers and not something else.

**Comment by DavidMacDonald** [comment]
My understanding is that the epub team has been looking to address this because it is not currently covered in WCAG. It's not covered in other SCs, because the page break locators are sometimes visible and sometimes not. They are programmatic markers (a bit of code) but they are often not made as **navigation** mechanisms. This sc says, if they are there turn the into navigation mechanisms to help blind and low vision users.

Epub simply wants a way to get to them if they are present.

**Comment by spanchang** [comment]
David writes: "the page break locators are sometimes visible and
sometimes not. They are programmatic markers (a bit of code) but they
do not automatically include a navigation mechanism for web browsers"
So, They are not available to anyone:
- If the locators are not visible or
- if they are not made as navigational mechanisms.
The page# in a Kindle / PDF reader is a user agent feature, is it not?
Should HTML Web page content author  be responsible for  replicating
that ?
The definition reads: "programmatic markers that are arranged in a
meaningful sequence to determine the location of a page in relation to
others in the set".
Does this cover something that is not navigable by anyone in a Web page?

Thanks,
Sailesh

On 10/11/21, Alastair Campbell ***@***.***> wrote:
> Proposed response (based on David's shorter comment):
>
> <hr>
>
> This SC was brought to the group by the EPUB group, who have been looking to
> address this because it is not currently covered in WCAG. It's not covered
> in other SCs, because the page break locators are sometimes visible and
> sometimes not. They are programmatic markers (a bit of code) but they do not
> automatically include a navigation mechanism for web browsers. This SC says,
> if they are present then include a navigation mechanisms to help blind and
> low vision users.
>
> Given the narrow scope of when this SC applies, the working group does not
> consider it an undue burden.
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/w3c/wcag/issues/2026#issuecomment-940194144


-- 
Sailesh Panchang
Principal Accessibility Consultant
Deque Systems Inc
381 Elden Street, Suite 2000, Herndon,
VA 20170
Mobile: 571-344-1765

**Comment by spanchang** [comment]
If this applies only to EPUB, is it better to add EPUB techniques  that link to existing SCs  instead of introducing a new SC?
An SC generally applies too all types of content not just a specific one.
This requirement is more along the lines of PDF techniques that help to make PDFs more accessible and meet existing SCs.  e.g.
PDF2: Creating bookmarks in PDF documents
PDF14: Providing running headers and footers in PDF documents
PDF17: Specifying consistent page numbering for PDF documents
Thanks,

**Comment by alastc** [comment]
@spanchang you appeared to have missed the discussion above, particular this [comment on the scenarios where it applies](https://github.com/w3c/wcag/issues/2026#issuecomment-938751663).

**Comment by alastc** [comment]
The response above was approved: https://www.w3.org/2021/12/03-ag-minutes.html#t12

Noting also that a clarification for the definition came out of the discussion: #2155
